RxJava module sporadic test failures

```
testObserveError(org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest)  Time elapsed: 0.643 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest.testObserveError(AsyncHttpObservableTest.java:98)

testObserveMultiple(org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest)  Time elapsed: 0.03 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest.testObserveMultiple(AsyncHttpObservableTest.java:121)

testObserveNoError(org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest)  Time elapsed: 0.007 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest.testObserveNoError(AsyncHttpObservableTest.java:78)

testToObservableError(org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest)  Time elapsed: 0.006 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest.testToObservableError(AsyncHttpObservableTest.java:58)

testToObservableNoError(org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest)  Time elapsed: 0.005 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.AsyncHttpObservableTest.testToObservableNoError(AsyncHttpObservableTest.java:38)

testAbort(org.asynchttpclient.extras.rxjava.single.AsyncHttpSingleTest)  Time elapsed: 0.012 sec  <<< FAILURE!
java.lang.AssertionError: Not completed!
    at rx.observers.TestSubscriber.assertCompleted(TestSubscriber.java:352)
    at org.asynchttpclient.extras.rxjava.single.AsyncHttpSingleTest.testAbort(AsyncHttpSingleTest.java:292)
```

--- a/extras/rxjava/src/test/java/org/asynchttpclient/extras/rxjava/AsyncHttpObservableTest.java
+++ b/extras/rxjava/src/test/java/org/asynchttpclient/extras/rxjava/AsyncHttpObservableTest.java
@@ -35,8 +35,8 @@ public class AsyncHttpObservableTest {
             o1.subscribe(tester);
             tester.awaitTerminalEvent();
             tester.assertTerminalEvent();
-            tester.assertCompleted();
             tester.assertNoErrors();
+            tester.assertCompleted();
             List<Response> responses = tester.getOnNextEvents();
             assertNotNull(responses);
             assertEquals(responses.size(), 1);
@@ -55,8 +55,8 @@ public class AsyncHttpObservableTest {
             o1.subscribe(tester);
             tester.awaitTerminalEvent();
             tester.assertTerminalEvent();
-            tester.assertCompleted();
             tester.assertNoErrors();
+            tester.assertCompleted();
             List<Response> responses = tester.getOnNextEvents();
             assertNotNull(responses);
             assertEquals(responses.size(), 1);
@@ -75,8 +75,8 @@ public class AsyncHttpObservableTest {
             o1.subscribe(tester);
             tester.awaitTerminalEvent();
             tester.assertTerminalEvent();
-            tester.assertCompleted();
             tester.assertNoErrors();
+            tester.assertCompleted();
             List<Response> responses = tester.getOnNextEvents();
             assertNotNull(responses);
             assertEquals(responses.size(), 1);
@@ -95,8 +95,8 @@ public class AsyncHttpObservableTest {
             o1.subscribe(tester);
             tester.awaitTerminalEvent();
             tester.assertTerminalEvent();
-            tester.assertCompleted();
             tester.assertNoErrors();
+            tester.assertCompleted();
             List<Response> responses = tester.getOnNextEvents();
             assertNotNull(responses);
             assertEquals(responses.size(), 1);
@@ -118,8 +118,8 @@ public class AsyncHttpObservableTest {
             all.subscribe(tester);
             tester.awaitTerminalEvent();
             tester.assertTerminalEvent();
-            tester.assertCompleted();
             tester.assertNoErrors();
+            tester.assertCompleted();
             List<Response> responses = tester.getOnNextEvents();
             assertNotNull(responses);
             assertEquals(responses.size(), 3);

--- a/extras/rxjava/src/test/java/org/asynchttpclient/extras/rxjava/single/AsyncHttpSingleTest.java
+++ b/extras/rxjava/src/test/java/org/asynchttpclient/extras/rxjava/single/AsyncHttpSingleTest.java
@@ -44,7 +44,6 @@ import org.testng.annotations.Test;
 import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
 import rx.Single;
@@ -99,8 +98,8 @@ public class AsyncHttpSingleTest {
 
         subscriber.awaitTerminalEvent();
         subscriber.assertTerminalEvent();
-        subscriber.assertCompleted();
         subscriber.assertNoErrors();
+        subscriber.assertCompleted();
         subscriber.assertValue(handler);
     }
 
@@ -152,8 +151,8 @@ public class AsyncHttpSingleTest {
 
         subscriber.awaitTerminalEvent();
         subscriber.assertTerminalEvent();
-        subscriber.assertCompleted();
         subscriber.assertNoErrors();
+        subscriber.assertCompleted();
         subscriber.assertValue(handler);
     }
 
@@ -289,13 +288,14 @@ public class AsyncHttpSingleTest {
                             return State.ABORT;
                         }
                     });
+
             underTest.subscribe(subscriber);
-            subscriber.awaitTerminalEvent(30, TimeUnit.SECONDS);
+            subscriber.awaitTerminalEvent();
         }
 
         subscriber.assertTerminalEvent();
-        subscriber.assertCompleted();
         subscriber.assertNoErrors();
+        subscriber.assertCompleted();
         subscriber.assertValue(null);
     }
 

